src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -127,7 +127,17 @@ public void ApplyDynamicallyAccessedMembersToType (ref ReflectionPatternContext
 			// Handle cases where a type has no members but annotations are to be applied to derived type members
 			reflectionPatternContext.RecordHandledPattern ();
 
-			MarkTypeForDynamicallyAccessedMembers (ref reflectionPatternContext, type, annotation, DependencyKind.DynamicallyAccessedMemberOnType);
+			// The annotations this type inherited from its base types should not produce
+			// warnings on the base members, since those are covered by warnings in the base type.
+			// For now, annotations that are inherited from base types but also declared explicitly on this type
+			// still warn about base members since the cache doesn't track enough information to tell the difference.
```

Comment:
I think this is fine - the only case where I can think of this making sense is if the derived type adds more annotations (base has all methods and derived type has both methods and fields) - but that is SO unlikely and rare - I don't care :-).